src/index.ts

```diff
@@ -955,12 +958,26 @@ export class BufferedChangeset implements IChangeset {
           const baseContent = this.safeGet(content, baseKey);
           const subContent = this.getDeep(baseContent, remaining.join('.'));
           const subChanges = getSubObject(changes, key);
-          // give back and object that can further retrieve changes and/or content
+          // give back an object that can further retrieve changes and/or content
           const tree = new ObjectTreeNode(subChanges, subContent, this.getDeep, this.isObject);
           return tree.proxy;
         } else if (typeof result !== 'undefined') {
           return result;
         }
+        const baseContent = this.safeGet(content, baseKey);
```

Comment:
What is the value of `result` here for the array case?  I thought one of the if expressions above would have been tripped.  It looks like there might be some minor refactors here.

Reply:
`result` is `undefined` here. It happens mostly when working with new array entries or previously deleted entries, iirc